csharp/Test/Integration/Memory/MemoryTest.cs

```diff
@@ -0,0 +1,311 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using TypeDB.Driver;
+using TypeDB.Driver.Api;
+using TypeDB.Driver.Common;
+using static TypeDB.Driver.Api.IThingType;
+using static TypeDB.Driver.Api.IThingType.Annotation;
+
+namespace TypeDB.Driver.Test.Integration
+{
+    public static class Utils
+    {
+        public static ITypeDBDriver OpenConnection()
+        {
+            ITypeDBDriver driver = TypeDB.CoreDriver(TypeDB.DEFAULT_ADDRESS);
+            Assert.IsNotNull(driver);
+            Assert.True(driver.IsOpen());
+
+            return driver;
+        }
+
+        public static void CloseConnection(ITypeDBDriver driver)
+        {
+            driver.Close();
+            Assert.False(driver.IsOpen());
+        }
+
+        public static void CreateDatabaseNoChecks(IDatabaseManager dbManager, string expectedDbName)
+        {
+            dbManager.Create(expectedDbName);
+        }
+
+        public static IDatabase CreateAndGetDatabase(
+            IDatabaseManager dbManager, string expectedDbName, bool checkAbsence = true)
+        {
+            if (checkAbsence)
+            {
+                Assert.False(dbManager.Contains(expectedDbName));
+            }
+
+            CreateDatabaseNoChecks(dbManager, expectedDbName);
+            Assert.True(dbManager.Contains(expectedDbName));
+
+            IDatabase db = dbManager.Get(expectedDbName);
+            Assert.IsNotNull(db);
+
+            string realDbName = db.Name;
+            Assert.AreEqual(expectedDbName, realDbName);
+
+            return db;
+        }
+
+        public static void DeleteDatabase(IDatabaseManager dbManager, IDatabase db)
+        {
+            string dbName = db.Name;
+            db.Delete();
+            Assert.False(dbManager.Contains(dbName));
+        }
+
+        public static void CheckAllDatabases(IDatabaseManager dbManager, ICollection<string> expectedDbNames)
+        {
+            var allDbs = dbManager.GetAll();
+            Assert.AreEqual(expectedDbNames.Count, allDbs.Count);
+
+            foreach (var db in allDbs)
+            {
+                Assert.True(expectedDbNames.Contains(db.Name));
+            }
+        }
+    }
+
+    [TestFixture]
+    public class MemoryTestFixture
```

Comment:
Sorry, i'm a bit confused - these are called 'memory' tests but i don't see anything to do with assertions around memory usage/layout/existence exactly? What is the intent of the test?

Reply:
All the memory manipulations are hidden within the logic and I don't really know what to check here regarding memory directly. I've named these tests `Memory` to check how marshalling of arrays and other complicated objects works via my FFI, and the tests' names had to highlight it: `SendArraysToNativeSide`, `SendOptionalEnumToNativeSide`. If something is wrong, it will not work / the data checks will fail / the tests will crash while running or stopping. 

I'm open to add comments / rename it.

Reply:
Maybe we can call it `MarshalTest` then? Just to use the vocabulary that you use yourself

Reply:
Okay!